From b2577ef7bd6285b3aa2a6b13269cad0938220c05 Mon Sep 17 00:00:00 2001
From: unknown <unknown@unknown>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 23/27] when using mke2fs, app can't move to SD Card On
 AndroidO, if enable mke2fs for vold, when moving apps to Sd Card,it would
 fail when creating Asec, because the param of block nums is wrong

On AndroidO MR1,vold used mke2fs by default, so apps can not be moved
to Sd Card. We should pass correct params to mke2fs.

Now ,i found another problem. When fix the wrong param above. Some apps
can be moved successfully, but other apps still failed. The reason is
"No space left on device" when uncompressing libs. But the size of
created ext4 image is larger than the size of apk and libs. The reason
is that when make ext4 image by mke2fs it need extra space for journal.
---
 fs/Ext4.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/Ext4.cpp b/fs/Ext4.cpp
index 927d4a3..0311f1f 100644
--- a/fs/Ext4.cpp
+++ b/fs/Ext4.cpp
@@ -211,7 +211,7 @@ status_t Format(const std::string& source, unsigned long numSectors,
     cmd.push_back(source);
 
     if (numSectors) {
-        cmd.push_back(StringPrintf("%lu", numSectors * (4096 / 512)));
+        cmd.push_back(StringPrintf("%lu", numSectors * 512 / 4096));
     }
 
     return ForkExecvp(cmd);
-- 
2.11.0

